src/k8s-extension/azext_k8s_extension/partner_extensions/OpenServiceMesh.py

```diff
@@ -93,3 +111,56 @@ def Update(self, extension, auto_upgrade_minor_version, release_train, version):
             release_train=release_train,
             version=version
         )
+
+
+def _validate_tested_distro(cmd, cluster_resource_group_name, cluster_name, extension_version):
+
+    field_unavailable_error = '\"testedDistros\" field unavailable for version {0} of Microsoft.openservicemesh, ' \
+        'cannot determine if this Kubernetes distribution has been properly tested'.format(extension_version)
+
+    if version.parse(str(extension_version)) <= version.parse("0.8.3"):
+        logger.warning(field_unavailable_error)
+        return
+
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+    resources = cf_resources(cmd.cli_ctx, subscription_id)
+
+    cluster_resource_id = '/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Kubernetes' \
+        '/connectedClusters/{2}'.format(subscription_id, cluster_resource_group_name, cluster_name)
+
+    resource = resources.get_by_id(cluster_resource_id, '2020-01-01-preview')
+    cluster_distro = resource.properties['distribution'].lower()
+
+    if cluster_distro == "general":
+        logger.warning('Unable to determine if distro has been tested for Microsoft.openservicemesh, '

```

Comment:
Is this capitalization correct? It should be `microsoft.openservicemesh` I think